### DISCUSSION TITLE
How can we retrieve the workspace name after the save operation in the deactivate() method?

### DISCUSSION THREAD

**Discussion by JoshwinThomasIBM** [microsoft/vscode-discussions#2458] [Extension Development QnA]
When the user chooses to save a workspace before closing the IDE, the deactivate method is triggered after the save operation. However, the workspace name appears empty because it hasn’t been saved yet. How can we retrieve the workspace name at this point?

**Comment by alefragnani** [comment]
Hi @JoshwinThomasIBM ,

I'm not sure you could accomplish what you wanted, but if not, I would suggest you to open a feature request in the VS Code repo (https://github.com/microsoft/vscode/issues), so the VS Code team could take a look. I suppose the WorkspaceFile could receive similar `onDid/onWill` Save events, similar to the ones available to `TextDocuments`.

I would suggest you to take a look at the `onDidChangeWorkspaceFolders` event as well. I know it is related to the _folders on that workspace_ but based on recent discussions about its behavior, maybe it may have a side effect based on the _first save_.

Hope this helps
